### DISCUSSION TITLE
Streamlining Navigation on Spina CMS

### DISCUSSION THREAD

**Discussion by EEUAAAIA** [SpinaCMS/Spina#1337] [Q&A] [answered]
I am working on enhancing the user experience on our website managed by Spina CMS.
The site features two sections: Preview and Gallery.

In the Preview section, each thumbnail is intended to redirect to the full page of the artwork. The code below currently generates these thumbnails:

```
<% art_preview = Spina::Page.live.where(view_template: :art_preview_items).last %>

 <% art_preview_items = Spina::Page.live.where(view_template: :art_preview_items) %>

<% art_preview_items.each do |art_preview_item| %>
  <a href="<%= art_preview_item.materialized_path %>">
    <% if art_preview.content.attachment_url(:image_1).present? && art_preview.content(:title_1).present? %>
      <div>
        <%= image_tag art_preview.content.attachment_url(:image_1) %>
        <div class="view"><%= t('view') %></div>
      </div>
      <div>
        <p><%= strip_tags(art_preview.content(:title_1)) %></p>
      </div>
    <% end %>
  </a>
<% end %>
```

Our image gallery is presented in a carousel format. The structure of our carousel gallery is as follows:

```
<div class="gallery__slider">
  <% Spina::Page.live.where(view_template: :art_gallery_items).each do |art_gallery_item| %>
    <% if art_gallery_item.content.attachment_url(:image).present? %>
      <a href="<%= art_gallery_item.materialized_path %>" >
        <%= image_tag art_gallery_item.content.attachment_url(:image) %>
        <div class="view"><%= t('view') %></div>
      </a>
    <% end %>
  <% end %>
</div>
```
When clicking on a work in the carousel, the user is directed to a distinct description page separate from the one in the Preview.

As a result, the artworks featured in the Preview section have two descriptive pages.

Our goal is to streamline this structure by ensuring that a click on a thumbnail in the Preview section directly redirects to the complete description page of the artwork within the Gallery.

(Perhaps using `materialized_path` or another routing method from Spina's system?)

I am open to your advice. Thank you in advance for your collaboration in this improvement."

**Comment by Bramjetten** [comment]
Have you taken a look at Spina::Resource?

**Reply by EEUAAAIA** [reply]
Yes, I try it but that was a fail : 

```
<% art_gallery_resource = Spina::Resource.find_by(view_template: :art_gallery_items) %>
  <% art_gallery_item = art_gallery_resource.find_by(slug: "art_gallery_items") %>
  <% if art_gallery_item %>
    <a href="<%= art_gallery_item.materialized_path %>">
      Test
    </a>
  <% else %>
    <p>Fail</p>
  <% end %>
```

My last shot was like that (Gaia is artwork's name): 

```
<% art_gallery = Spina::Page.live.where(view_template: :art_gallery_items) %>
<% art_gallery.each do |art_gallery_item| %>
  <% if art_gallery_item.title == "Gaia" %>
  <%= link_to "Gaia", art_gallery_item.materialized_path %>
  <% else %>
    <p>Échec</p>
  <% end %>
<% end %>
```

**Comment by Bramjetten** [comment]
Thats not how resources work in Spina. You can think of a resource as a separate collection of pages. You can then reference those pages elsewhere.

**Comment by bolom** [comment, marked as answer]
@EEUAAAIA 

```
pages = Spina::Resource.find_by(name:  "art_gallery_items").pages
pages.joins(:translations).where(translations: { title: "test" })
```
